crates/rpc/rpc-types/src/otterscan.rs

```diff
@@ -0,0 +1,72 @@
+use crate::{Block, Transaction, TransactionReceipt};
+use reth_primitives::{Address, Bytes};
+use serde::{Deserialize, Serialize};
+
+/// Operation type enum for `InternalOperation` struct
+#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+pub enum OperationType {
+    /// Operation Transfer
+    OpTransfer,
+    /// Operation Contract self destruct
+    OpSelfDestruct,
+    /// Operation Create
+    OpCreate,
+    /// Operation Create2
+    OpCreate2,
+}
+
+/// Custom struct for otterscan `getInternalOperations` RPC response
+#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+pub struct InternalOperation {
+    r#type: OperationType,
+    from: Address,
+    to: Address,
+    value: u128,
+}
+
+/// Custom struct for otterscan `traceTransaction` RPC response
+#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+pub struct TraceEntry {
+    r#type: String,
+    depth: u32,
+    from: Address,
+    to: Address,
+    value: u128,
+    input: Bytes,
+}
+
+/// Internal issuance struct for `BlockDetails` struct
+#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct InternalIssuance {
+    block_reward: String,
+    uncle_reward: String,
+    issuance: String,
+}
+
+/// Custom struct for otterscan `getBlockDetails` RPC response
+#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct BlockDetails {
+    block: Block,
+    issuance: InternalIssuance,
+    total_fees: u64,
+}
+
+/// Custom struct for otterscan `searchTransactionsAfter`and `searchTransactionsBefore` RPC
+/// responses
+#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct TransactionsWithReceipts {
+    txs: Vec<Transaction>,
+    receipts: TransactionReceipt,
```

Comment:
this should be a Vec

but more importantly, I'm finding now (while working on anvil) that the response otterscan expects doesn't quite match what their spec says, 

e.g. for `ots_getBlockTransactions`, they apparently expect something like:
```
{
fullblock: { transactions: [ ... ], transactionCount: xx, ... /* all the other usual fields */ },
receipts: [ ... ]
}
```

overall, maybe it's better to hold off on these structs and instead add them 1 by 1 as we figure out these inconsistencies
wdyt @mattsse ?

Reply:
My bad ! Should have doubled checked this. 
We really need to write some tests , with some request responses then check that the serializations and deserializations matches up.